test/fft_test.rg

```diff
@@ -17,92 +17,415 @@ import "regent"
+local compare_regions_4d_complex64 = make_compare_regions_task(region(ispace(int4d), complex64), region(ispace(int4d), complex64))
+
+-- INTERFACES
+
+local fft1d_float_complex32 = fft.generate_fft_interface(int1d, float, complex32)
+local fft1d_complex32_complex32 = fft.generate_fft_interface(int1d, complex32, complex32)
+
+local fft1d_double_complex64 = fft.generate_fft_interface(int1d, double, complex64)
+local fft1d_complex64_complex64 = fft.generate_fft_interface(int1d, complex64, complex64)
+local fft2d_complex64_complex64 = fft.generate_fft_interface(int2d, complex64, complex64)
+local fft3d_complex64_complex64 = fft.generate_fft_interface(int3d, complex64, complex64)
+
+local fft2d_batch_double_complex64 = fft.generate_fft_interface(int3d, double, complex64)
+local fft2d_batch_complex64_complex64 = fft.generate_fft_interface(int3d, complex64, complex64)
+local fft3d_batch_complex64_complex64 = fft.generate_fft_interface(int4d, complex64, complex64)
+
+-- TEST FUNCTIONS
 
 __demand(__inline)
-task test1d()
-  var r = region(ispace(int1d, 128), complex64)
-  var s = region(ispace(int1d, 128), complex64)
-  fill(r, 0)
+task test_1d_float_to_complex32_transform()
+  format.println(">> test_1d_float_to_complex32_transform")
+
+  var r = region(ispace(int1d, 3), float)
+  var s = region(ispace(int1d, 3), complex32)
+  var p = region(ispace(int1d, 1), fft1d_float_complex32.plan)
+
+  fill(r, 3)
   fill(s, 0)
-  var p = region(ispace(int1d, 1), fft1d.plan)
-  -- Important: this overwrites r and s!
-  fft1d.make_plan(r, s, p)
-  fill(r, 0)
+
+  fft1d_float_complex32.make_plan(r, s, p)
+  fft1d_float_complex32.execute_plan(r, s, p)
+  fft1d_float_complex32.destroy_plan(p)
+
+  -- Verify
+  var e = region(ispace(int1d, 3), complex32)
+  fill(e, 0)
+  e[0].real = 9
+
+  print_region_1d_float("Input", r)
+  print_region_1d_complex32("Output", s)
+  print_region_1d_complex32("Expected", e)
+
+  var status = compare_regions_1d_complex32(s, e)
+  format.println("<< test_1d_float_to_complex32_transform [{}]", status)
+end
+
+__demand(__inline)
+task test_1d_complex32_to_complex32_transform()
+  format.println(">> test_1d_complex32_to_complex32_transform")
+  var r = region(ispace(int1d, 3), complex32)
+  var s = region(ispace(int1d, 3), complex32)
+  var p = region(ispace(int1d, 1), fft1d_complex32_complex32.plan)
+
+  for x in r do
+    r[x].real = 3
+    r[x].imag = 3
+  end
   fill(s, 0)
-  fft1d.execute_plan(r, s, p)
-  fft1d.destroy_plan(p)
+
+  fft1d_complex32_complex32.make_plan(r, s, p)
+  fft1d_complex32_complex32.execute_plan(r, s, p)
+  fft1d_complex32_complex32.destroy_plan(p)
+
+  -- Verify
+  var e = region(ispace(int1d, 3), complex32)
+  fill(e, 0)
+  e[0].real = 9
+  e[0].imag = 9
+
+  print_region_1d_complex32("Input", r)
+  print_region_1d_complex32("Output", s)
+  print_region_1d_complex32("Expected", e)
+
+  var status = compare_regions_1d_complex32(s, e)
+  format.println("<< test_1d_complex32_to_complex32_transform [{}]", status)
```

Comment:
Like I said above I think we should make the `status` here a boolean. Also, this test as written will not actually fail the process if it does not pass. We need an assert so that a failing test will actually fail the process. Otherwise we'll never see the failures. It's still ok to print if you want, we just need the assert to see the failure.

Reply:
Added a boolean and asserts: https://github.com/arjunkunna/regent-fft/commit/686a130f6f12c8154a24511b58248d6a22506470